src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,35 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                let check_lock_res =
+                    self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock));
+                if let Err(e) = check_lock_res {
+                    *can_be_cached.unwrap() = can_be_cached_by_lock;
+                    return Err(e);
```

Comment:
Only a conflict lock error can be cached. Other errors like getting snapshot error shouldn't be cached.

Reply:
Can be cached or not is decided by load_and_check_lock and return in can_be_cached_by_lock,
the code here is not deciding cache_able, just assigning can_be_cached before return.
(Ugly, though)